homeassistant/components/netgear/router.py

```diff
@@ -0,0 +1,283 @@
+"""Represent the Netgear router and its devices."""
+from datetime import timedelta
+import logging
+
+from pynetgear import Netgear
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_SSL,
+    CONF_USERNAME,
+)
+from homeassistant.core import callback
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC, format_mac
+from homeassistant.helpers.dispatcher import (
+    async_dispatcher_connect,
+    async_dispatcher_send,
+)
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import async_track_time_interval
+from homeassistant.helpers.typing import HomeAssistantType
+from homeassistant.util import dt as dt_util
+
+from .const import CONF_CONSIDER_HOME, DEFAULT_CONSIDER_HOME, DOMAIN, MODELS_V2
+from .errors import CannotLoginException
+
+SCAN_INTERVAL = timedelta(seconds=30)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def get_api(
+    password: str,
+    host: str = None,
+    username: str = None,
+    port: int = None,
+    ssl: bool = False,
+) -> Netgear:
+    """Get the Netgear API and login to it."""
+    api: Netgear = Netgear(password, host, username, port, ssl)
+
+    if not api.login():
+        raise CannotLoginException
+
+    return api
+
+
+async def async_setup_netgear_entry(
+    hass: HomeAssistantType,
+    entry: ConfigEntry,
+    async_add_entities,
+    entity_class_generator,
+) -> None:
+    """Set up device tracker for Netgear component."""
+    router = hass.data[DOMAIN][entry.unique_id]
+    tracked = set()
+
+    @callback
+    def update_router():
+        """Update the values of the router."""
+        add_entities(router, async_add_entities, tracked, entity_class_generator)
+
+    entry.async_on_unload(
+        async_dispatcher_connect(hass, router.signal_device_new, update_router)
+    )
+
+    update_router()
+
+
+@callback
+def add_entities(router, async_add_entities, tracked, entity_class_generator):
+    """Add new tracker entities from the router."""
+    new_tracked = []
+
+    for mac, device in router.devices.items():
+        if mac in tracked:
+            continue
+
+        new_tracked = new_tracked + entity_class_generator(router, device)
```

Comment:
```suggestion
        new_tracked.extend(entity_class_generator(router, device))
```

Will this do what you need?

Reply:
yes